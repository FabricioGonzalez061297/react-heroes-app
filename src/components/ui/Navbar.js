import React, { useContext } from 'react'
import { Link, NavLink, useHistory } from 'react-router-dom'
import { AuthContext } from '../../auth/AuthContext';
import { types } from '../../types/types';

export const Navbar = () => {

    const {user:{name},dispatch} =useContext(AuthContext);
    const history = useHistory();


    const handleLogout=()=>{
        dispatch({type:types.logout});
        history.replace('/login')
    }

    return (
        <nav  className="navbar navbar-expand-sm navbar-dark bg-dark">
            
            <Link  style={{marginLeft:'.5rem'}}
                className="navbar-brand" 
                to="/"
            >
                Asociaciones
            </Link>

            <div className="navbar-collapse">
                <div className="navbar-nav">

                    <NavLink 
                        activeClassName="active"
                        className="nav-item nav-link" 
                        exact
                        to="/marvel"
                    >
                        Marvel
                    </NavLink>

                    <NavLink 
                        activeClassName="active"
                        className="nav-item nav-link" 
                        exact
                        to="/dc"
                    >
                        DC
                    </NavLink>
                    <NavLink 
                        activeClassName="active"
                        className="nav-item nav-link" 
                        exact
                        to="/search"
                    >
                        Search
                    </NavLink>
                </div>
            </div>

            {/* <div className="navbar-collapse collapse w-100 order-3 dual-collapse2"> */}
                <ul className="navbar-nav ">

                    <span className="nav-item nav-link text-info "> {name} </span>
                    <button 
                        style={{marginRight:'.5rem'}}
                        onClick={handleLogout}
                        className="nav-item nav-link btn " 
                        type="button"
                    >
                        Logout
                    </button>
                </ul>
            {/* </div> */}
        </nav>
    )
}