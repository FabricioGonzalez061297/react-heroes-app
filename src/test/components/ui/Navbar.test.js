import { mount } from "enzyme";
import { MemoryRouter,Router } from "react-router-dom";
import { AuthContext } from "../../../auth/AuthContext";
import { Navbar } from "../../../components/ui/Navbar";
import { types } from "../../../types/types";

describe('pruebas en <Navbar/>', () => {

    const historyMock = {
        push: jest.fn(),
        location: {},
        listen:jest.fn(),
        createHref:jest.fn(),
        replace:jest.fn()
    }

    const authContext = {
        dispatch: jest.fn(),
        user: {
            logged: true,
            name: 'fabricio'
        }
    }

    const wrapper = mount(
        <AuthContext.Provider value={authContext} >
            <MemoryRouter>
                <Router history={historyMock} >
                    <Navbar />
                </Router>

            </MemoryRouter>
        </AuthContext.Provider>
    );


    afterEach(()=>{
        jest.clearAllMocks();
    })

    test('debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.text-info').text().trim()).toBe('fabricio');
    });

    test('debe de llamar el logaout y llamar histori', () => {
        wrapper.find('button').prop('onClick')();

        expect(authContext.dispatch).toHaveBeenLastCalledWith({ type: types.logout });
        expect(historyMock.replace).toHaveBeenCalledWith('/login');

    });



});
