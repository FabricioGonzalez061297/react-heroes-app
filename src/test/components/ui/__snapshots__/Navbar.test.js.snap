// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <Navbar/> debe de mostrarse correctamente 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    onClick={[Function]}
    style={
      Object {
        "marginLeft": ".5rem",
      }
    }
  >
    Asociaciones
  </a>
  <div
    className="navbar-collapse"
  >
    <div
      className="navbar-nav"
    >
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        Marvel
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        DC
      </a>
      <a
        aria-current={null}
        className="nav-item nav-link"
        onClick={[Function]}
      >
        Search
      </a>
    </div>
  </div>
  <ul
    className="navbar-nav "
  >
    <span
      className="nav-item nav-link text-info "
    >
       
      fabricio
       
    </span>
    <button
      className="nav-item nav-link btn "
      onClick={[Function]}
      style={
        Object {
          "marginRight": ".5rem",
        }
      }
      type="button"
    >
      Logout
    </button>
  </ul>
</nav>
`;
